### DISCUSSION TITLE
Map fields with @id decorator to String

### DISCUSSION THREAD

**Discussion by georgekrax** [graphql-nexus/nexus-prisma#173] [Q&A]
Hello everyone,

I would like to force the **Prisma Client** to generate `string` instead of `number` for `Int` fields with @id. Is there a way to do it? I have a GraphQL API that treats `id` fields, with the corresponding string **GraphQL scalar ID** type, but TypeScript throws an error because the Client parses `@id` to `number`, and I do not want to replace `Int @id` with `String @id`. 

Thanks in advance!

**Comment by Frosty21** [comment]
It should automatically set `id: Scalars["ID"]` ID: as string in your generated graphql.tsx
```
export type Scalars = {
	ID: string;
	String: string;
	Boolean: boolean;
	Int: number;
	Float: number;
	/** BigInt custom scalar type */
	BigInt: any;
	/** Date custom scalar type */
	DateTime: any;
	/** Decimal custom scalar type */
	Decimal: any;
	/** Json custom scalar type */
	Json: any;
};
```
check your schema.prisma file if you want a guid its should be `  id             String      @id @default(uuid())
` for example
Let me know if that helps or you need more help we can chat further

**Reply by georgekrax** [reply]
I use `Int` or `BigInt` for id fields on my `schema.prisma` file. It is settled up like this:
```prisma
model User {
  id Int @id @default(autoincrement())
}
```
